AdobeStockImage/Model/SaveImagePreview.php

```diff
@@ -208,6 +227,21 @@ private function getImagesByAdobeId(int $adobeId): AssetSearchResultsInterface
             ->setSortOrders([])
             ->create();
 
-        return $this->getImageList->execute($searchCriteria);
+        $searchResult = $this->getImageList->execute($searchCriteria);
+
+        $items = [];
+        /** @var Document $item */
+        foreach ($searchResult->getItems() as $item) {
```

Comment:
As finally only one asset is needed here, the algorithm can be simplified. only the first item can be converted and returned if it exists (no need for the foreach loop and searchResult creation).

In this case the method should be renamed to `getImageByAdobeId` and return AssetInterface.

`execute` method should be adjusted accordingly